tests/Integration/Behaviour/Features/Context/Domain/Carrier/CarrierFeatureContext.php

```diff
@@ -292,11 +286,6 @@ public function assertCarrierProperties(string $reference, TableNode $tableNode)
             );
         }
 
-        if (isset($data['taxRuleGroup'])) {
```

Comment:
I think the assertion part can remain in this step, especially since the tax rules validation in the other context is still based on the same Query, it would be simpler to keep all the assertions in one place